src/selectors/profile.js

```diff
@@ -120,6 +121,10 @@ export const getFocusCallTreeGeneration: Selector<number> = (state) =>
   getProfileViewOptions(state).focusCallTreeGeneration;
 export const getZeroAt: Selector<Milliseconds> = (state) =>
   getProfileRootRange(state).start;
+export const getProfileTimelineUnit: Selector<string> = (state) => {
+  const { sampleUnits } = getProfile(state).meta;
+  return sampleUnits ? sampleUnits.time : '';
+};
```

Comment:
So this should then return the new `TimelineUnit` type and default to `"ms"`.

Reply:
Oh and the existing type `SampleUnits` needs to be changed so that, instead of requiring `"ms"` in the `time` field, it accepts a `TimelineUnit`. Otherwise the `getProfileTimelineUnit` function isn't meaningful if the type requires that it's always `"ms"` anyway.